easybuild/easyconfigs/p/PostgreSQL/PostgreSQL-9.6.2-intel-2017a-Python-2.7.13.eb

```diff
@@ -0,0 +1,46 @@
+easyblock = 'ConfigureMake'
+
+name = 'PostgreSQL'
+version = '9.6.2'
+versionsuffix = '-Python-%(pyver)s'
+
+homepage = 'https://www.postgresql.org/'
+description = """PostgreSQL is a powerful, open source object-relational database system.
+ It is fully ACID compliant, has full support for foreign keys,
+ joins, views, triggers, and stored procedures (in multiple languages).
+ It includes most SQL:2008 data types, including INTEGER,
+ NUMERIC, BOOLEAN, CHAR, VARCHAR, DATE, INTERVAL, and TIMESTAMP.
+ It also supports storage of binary large objects, including pictures,
+ sounds, or video. It has native programming interfaces for C/C++, Java,
+ .Net, Perl, Python, Ruby, Tcl, ODBC, among others, and exceptional documentation."""
+
+toolchain = {'name': 'intel', 'version': '2017a'}
+
+source_urls = ['http://ftp.postgresql.org/pub/source/v%(version)s/']
+sources = [SOURCELOWER_TAR_GZ]
+
+dependencies = [
+    ('libreadline', '6.3'),
```

Comment:
@tutufan The problem with the conflict seems to be here: you're including `libreadline` 6.3 as dependency for `PostgreSQL`, while `Python` 2.7.13 depends on `libreadline` 7.0 (see https://github.com/easybuilders/easybuild-easyconfigs/blob/master/easybuild/easyconfigs/p/Python/Python-2.7.13-intel-2017a.eb#L18).

I'm not sure how this is not an issue for you... Doesn't your `Python/2.7.13-intel-2017a` module depend on `libreadline` 7.0?

The `Python-2.7.13-intel-2017a.eb` in the central repository has always depended on `libreadline` 7.0, see #4260

Reply:
Hmm.  Not sure how this is supposed to work.  Octave depends on PostgreSQL, which in turn depends on Python.  In addition, all three directly depend on libreadline.  Octave depends on 6.3, PostgreSQL depends on 6.3, and Python depends on 7.0.  (I'm reading these deps from the lua files.)

And just to be extra annoying, `lmod` is happily sailing through this without a complaint.  It loads only 7.0.

One wrinkle that might matter is that I probably built all three of these modules in separate eb runs.  That is, I don't think that they weren't built as a result of a `--robot` dep.

How exactly does `eb` come up with the dep graph that it checks for consistency?  I'd like it to use the deps from the already-built modules that are being used in the build (together with deps for any robot modules it plans to build).  But it could also be using the deps from the easyconfig tree, and theoretically those could be somewhat out of sync with the built modules, since I have been occasionally switching from branch to branch (master, develop, and my topic branches).

I'll try to attach a log from the Octave build.  It's certainly possible that I've confused myself somehow, but not sure why neither `eb` nor `lmod` are scolding me.
[easybuild-Octave-4.2.1-20180122.132842.log.txt](https://github.com/easybuilders/easybuild-easyconfigs/files/1658035/easybuild-Octave-4.2.1-20180122.132842.log.txt)